kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/BrokersController.java

```diff
@@ -41,4 +43,11 @@
   ) {
     return Mono.just(ResponseEntity.ok(clusterService.getAllBrokersLogdirs(clusterName, brokers)));
   }
+
+  @Override
+  public Mono<ResponseEntity<BrokerLogdirUpdateResult>> updateBrokerTopicPartitionLogDir(
+      String clusterName, Integer id, Mono<BrokerLogdirUpdate> brokerLogdir,
+      ServerWebExchange exchange) {
+    return clusterService.updateBrokerLogDir(clusterName, id, brokerLogdir).map(ResponseEntity::ok);
+  }
```

Comment:
Timur, could you please refactor your method not to pass Mono/Flux objects as a function parameter? For example, just using
`brokerLogDir.map(... -> clusterService.updateBrokerLogDir(...))` instead?
I know there're a lot of places where we've done it in such a way.

Reply:
Fixed, please review

Reply:
To be precise, it's not always Bad Request, we should provide 5xx errors also, and maybe it's better to throw on Mono level, to catch and provide as other errors

Reply:
Fixed